src/table/table-adapter.class.ts

```diff
@@ -151,6 +186,53 @@ export class TableDomAdapter implements TableAdapter {
 	 */
 	findColumnIndex(cell: HTMLTableCellElement): number {
 		const row = this.getRow(this.findRowIndex(cell));
+		// if the cell has linked headers we can do a more accurate lookup
+		if (cell.headers) {
+			const ids = cell.headers.split(" ");
+			const headerRows = Array.from(this.tableElement.tHead.rows);
+			const indexes = [];
+
+			// start from the last row and work up
+			for (const headerRow of headerRows.reverse()) {
+				const cells = Array.from(headerRow.cells);
+				const header = cells.find(cell => ids.includes(cell.id));
+				// if we have a matching header, find it's index (adjusting for colspans)
+				if (header) {
+					// this is borrowed from below
+					let cellIndex = 0;
+					for (const c of cells) {
+						if (c === header) { break; }
+						cellIndex += c.colSpan;
+					}
+					indexes.push(cellIndex);
+				}
+			}
+
+			// sort the indexes largest to smallest to find the closest matching header index
+			const firstIndex = indexes.sort((a, b) => b - a)[0];
+
+			// search the row for cells that share the header
+			let similarCells = [];
+			for (const id of ids) {
+				// there's no selector that will match two space separated lists,
+				// so we have to iterate through the ids and query the row for each
+				const cells = Array.from(row.querySelectorAll(`[headers~='${id}']`));
+				for (const cell of cells) {
```

Comment:
Does `cell` not clash here with the `cell` parameter in the function?

Reply:
Huh, totally missed that ... `const`/`let` will lexically scope it so it's not an issue, but for readability we should rename it 👍